torch/csrc/distributed/c10d/init.cpp

```diff
@@ -605,6 +658,58 @@ They are used in specifying strategies for reduction collectives, e.g.,
       py::arg("buffer_size"),
       py::call_guard<py::gil_scoped_release>());
 
+  module.def(
```

Comment:
Why this test is implemented in `init.cpp` instead of in a cpp test file? It looks a little weird to have a test here.

Reply:
The test needs a Python class to properly test the trampoline. I don't think we have a separate Python shared object with test-only code, so this is the next best thing. Not ideal, but not too bad either, IMO.